### ISSUE TITLE
Accessibility for Embedded Video in Wagtail

### ISSUE THREAD

**Issue by dbietila** [uchicago-library/library_website#633]
We received this message from CDA regarding a video posted on our site as part of a web exhibit:

We noticed that the Library's Siteimprove score dropped about 7 points recently. This was due to a single silent video on this [Paul Moses: New Beginnings](https://www.lib.uchicago.edu/collex/exhibits/paul-b-moses-trailblazing-art-historian/new-beginnings/) special exhibition webpage triggering four potential issues for review, each of which is worth 1.91 points.

1.	Is this video audio-described?
2.	Does this video have captions?
3.	Is there an alternative to the visual content in this video?
4.	Do visual-only videos have an accessible alternative?

This series of issues is triggered specifically when silent video clips are added. A similar situation occurred on the Academic Technologies site, so we wanted to reach out with recommendations for how to resolve.

Also, we would like to start a discussion around the CDA manually assessing some pages on the Library site. I'll send a separate message with more information.

Video Recommendations:

The silent 7-minute home movie clip of Paul Moses and Alice Johnson’s wedding has a couple issues currently--it is missing [captions and an alternative to the video-only content](https://digitalaccessibility.uchicago.edu/resources/content-creators/captions-transcripts-audio-description/).

1) Add an aria-label attribute to the <video> tag and then include a simple closed caption track to the video that says "silent throughout" or similar. Attached is an example you could use after editing the timestamps.

```
<video width="320" height="240" controls="controls" aria-label="Footage of Paul Moses and Alice Johnson’s wedding, 1962">
     <source src="/media/media/Moses_Wedding_Video_1_9N124Q9.mp4" type="video/mp4">
     <track srclang="en" kind="captions" src="captions.vtt" type="text/vtt" default=""></track>
</video>
```

2) Since the video is silent, it should also have either a [text alternative or an audio track that presents equivalent information](https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded.html). A text alternative could appear or be linked to below the video. For creating an alternate audio track, 3Play Media offers that as a service. Their [UChicago rate](https://digitalaccessibility.uchicago.edu/resources/content-creators/3play-media/) for audio descriptions is $8.50 per minute, so you'd be looking at something around $60 for this video. Whichever method you choose and how to describe the video is up to you.

Once those two items are added to the video the 4 potential issues can be cleared in Siteimprove, which will earn back 7.64 points the next time the site is crawled.

**Comment by dbietila** [comment]
Currently, the code in Wagtail is:
```
<video controls="" width="320" height="240">
                    <source src="/media/media/Moses_Wedding_Video_1_9N124Q9.mp4" type="video/mp4">
                    Your browser does not support the video tag.
                </video>
```
1) It looks like we'll need a means of adding an aria-label attribute to the video element. 
2) We'll also need a way to add the <track> element to the video block.
3) It might be nice to have an official way to link to a text alternative for the video, but I'm sure we can kludge something together using existing Wagtail functionality.
